pkg/sdkserver/sdkserver.go

```diff
@@ -144,7 +144,7 @@ func NewSDKServer(gameServerName, namespace string, kubeClient kubernetes.Interf
 		s1 := fields.OneTermEqualSelector("metadata.name", gameServerName)
 		opts.FieldSelector = s1.String()
 	}
-	factory := externalversions.NewSharedInformerFactoryWithOptions(agonesClient, 30*time.Second, externalversions.WithNamespace(namespace), externalversions.WithTweakListOptions(tweakListOptions))
+	factory := externalversions.NewSharedInformerFactoryWithOptions(agonesClient, 0, externalversions.WithNamespace(namespace), externalversions.WithTweakListOptions(tweakListOptions))
```

Comment:
I've always wondered if 0 was "never" - did you find  a doc on that? I never managed to actually find an answer either way.

Reply:
https://github.com/kyverno/kyverno/issues/2792
https://github.com/kyverno/kyverno/pull/2806/files

They gave it a shot with 0, but then they decided they didn't need it anymore and closed it.

Reply:
https://firehydrant.com/blog/stay-informed-with-kubernetes-informers/

Reply:
Took me a bit, but I found the code! You are correct - 0 is _off_.

https://github.com/kubernetes/client-go/blob/84a6fe7e4032ae1b8bc03b5208e771c5f7103549/tools/cache/shared_informer.go#L283-L286

Reply:
Shall we keep this PR and close the other PR?

Reply:
Let's keep both running - curious to see what the e2e results are.

I'm leaning more towards this one, but let's see what happens.

Reply:
quick question:
Informers keep an eye on what's happening in Kubernetes resources and keeping the local cache up to date with changes happening in the cluster, right? 
Disable resync will be helpful when there are no resource changes, but when there are changes in the resources how do informers function?

Reply:
> Informers keep an eye on what's happening in Kubernetes resources and keeping the local cache up to date with changes happening in the cluster, right?

100% nailed it.

> Disable resync will be helpful when there are no resource changes, but when there are changes in the resources how do informers function?

Informers use a Kuberntes watch operations to keep up to date with changes that have happened in the K8s cluster.

See: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#:~:text=Watch%3A%20Watch%20will%20stream%20results%20for%20an%20object(s)%20as%20it%20is%20updated.%20Similar%20to%20a%20callback%2C%20watch%20is%20used%20to%20respond%20to%20resource%20changes.

What resync does it forces a full list of all the K8s resources rather than just pushing changes to the Informer.

Next steps:
Let's close the other PR, since this pass e2e tests - and let's move this change behind a Feature Flag, so we can test it with it enabled and disabled, and see the performance impact!

We can do this all in this PR if that's easiest for you.

So we'll need a new feature flag in here:

Reply:
Updated above comment - hit submit too early.

Reply:
Closed the other PR.
Followed these steps: https://github.com/googleforgames/agones/blob/25e0c4a3e85688d181b5f155d1ef66b576d839bc/pkg/util/runtime/features.go#L77-L83

I am confused with true and false [here](https://github.com/googleforgames/agones/blob/25e0c4a3e85688d181b5f155d1ef66b576d839bc/pkg/util/runtime/features.go#L82)😅. Please review and share your thoughts👍🏻